pages/search.tsx

```diff
@@ -0,0 +1,188 @@
+  searchState: SearchState;
+  searchPhrase: string;
+}
+
+const SearchState = ({ searchState, searchPhrase }: SearchStateProps) => {
+  if (searchState.type === "INIT") {
+    return (
+      <p className="text-fpl-purple">
+        Search results will appear here. You can search by name or team name
+      </p>
+    );
+  }
+  if (searchState.type === "EMPTY") {
+    return (
+      <p className="text-fpl-purple">
+        Please enter a search value. You can search by name or team name
+      </p>
+    );
+  }
+  if (searchState.type === "ERROR") {
+    return (
+      <p className="text-fpl-purple">
+        Ooops, looks like something went wrong 🤕
+      </p>
+    );
+  }
+  if (searchState.type === "SUCCESS") {
+    if (searchState.data.length < 1) {
+      return (
+        <p className="text-fpl-purple">
+          Search for "{searchPhrase}" did not return any matches
+        </p>
+      );
+    }
+    return <ResultTable playerEntries={searchState.data} searchPhrase={searchPhrase}/>
+    
+  }
+};
+
+interface ResultTableProps {
+  playerEntries: PlayerEntry[];
+  searchPhrase: string;
+}
+
+const ResultTable = ({ playerEntries, searchPhrase }: ResultTableProps) => {
+  return (
+    <div className="w-full md:w-3/6 m-auto">
+      <p className="text-fpl-purple text-xl md:text-xl text-left">
+        Search results for "{searchPhrase}"
+      </p>
+      <table className="w-full flex flex-row flex-no-wrap rounded overflow-hidden sm:shadow-lg my-5">
+        <thead className="text-white">
+          {playerEntries.map((data, i) => (
+            <tr className="bg-fpl-purple flex flex-col flex-no wrap sm:table-row rounded-l-lg sm:rounded-none mb-2 sm:mb-0" key={`table-header-${i}`}>
+              <th className="p-3 text-left">Name</th>
+              <th className="p-3 text-left">Team name</th>
+              <th className="p-3 text-left" style={{ width: "120px" }}>
+                Open team
+              </th>
+            </tr>
+          ))}
+        </thead>
+        <tbody className="flex-1 sm:flex-none">
+          {playerEntries.map((data, i) => (
+            <tr className="flex flex-col flex-no wrap sm:table-row mb-2 sm:mb-0 bg-white rounded-r-lg sm:rounded-none" key={`table-row-${i}`}>
+              <td className="text-left border-grey-light border hover:bg-gray-100 p-3">
+                {data.realName}
+              </td>
+              <td className="text-left border-grey-light border hover:bg-gray-100 p-3 truncate">
+                {data.teamName}
+              </td>
+              <td className="text-left md:text-center border-grey-light border hover:bg-gray-100 p-3 text-red-400 hover:text-red-600 hover:font-medium cursor-pointer">
+                <a
+                  href={`https://fantasy.premierleague.com/entry/${data.entry}/history`}
+                  className="block"
+                  target="_blank"
+                >
+                  ➡️
+                </a>
+              </td>
```

Comment:
I've added a `verifiedEntryEmoji` property now, which currently just holds a chess pawn for Carlsen. Could maybe fit that in somewhere? 😉